src/test/regress/input/temp_tablespaces.source

```diff
@@ -301,3 +301,246 @@ $$;
 drop table tmp_table_12871;
 drop tablespace space_12871;
 
+--
+-- temp_spill_files_tablespaces:
+--
+
+-- start_ignore
+DROP TABLE IF EXISTS tts_foo;
+DROP TABLE IF EXISTS tts_bar;
+DROP TABLE IF EXISTS tts_jazz;
+DROP TABLESPACE IF EXISTS temp_tables;
+DROP TABLESPACE IF EXISTS temp_files0;
+DROP TABLESPACE IF EXISTS temp_files1;
+DROP TABLESPACE IF EXISTS temp_files2;
+DROP TABLESPACE IF EXISTS temp_files3;
+-- end_ignore
+
+CREATE TABLE tts_foo (a int, b int) DISTRIBUTED BY (a);
+CREATE TABLE tts_bar (c int, d int) DISTRIBUTED BY (d);
+CREATE TABLE tts_jazz (e int, f int) DISTRIBUTED BY (e);
+
+INSERT INTO tts_foo SELECT i, i FROM generate_series (1, 1000000) i;
+INSERT INTO tts_bar SELECT i, i FROM generate_series (6000, 120000) i;
+INSERT INTO tts_jazz SELECT i, i FROM generate_series (4000, 150000) i;
+
+ANALYZE tts_foo;
+ANALYZE tts_bar;
+ANALYZE tts_jazz;
+
+CREATE TABLESPACE temp_tables LOCATION '@testtablespace@_temp_tables';
+CREATE TABLESPACE temp_files0 LOCATION '@testtablespace@_temp_files0';
+CREATE TABLESPACE temp_files1 LOCATION '@testtablespace@_temp_files1';
+CREATE TABLESPACE temp_files2 LOCATION '@testtablespace@_temp_files2';
+CREATE TABLESPACE temp_files3 LOCATION '@testtablespace@_temp_files3';
+
+RESET temp_tablespaces;
+RESET temp_spill_files_tablespaces;
+
+-- Creates a temporary table and remembers the tablespace it was created in.
+-- Runs a query that creates some temporary files, calculates the tablespace for
+-- temporary files that will be used for current session, and checks if any
+-- files were created in that tablespace.
+--
+-- 'n_temp_spill_files_tablespaces' is the count of tablespaces specified in
+-- 'temp_spill_files_tablespaces', either global or session-wide.
+CREATE OR REPLACE FUNCTION gp_tablespace_file_report(
+    n_temp_spill_files_tablespaces int
+) RETURNS TABLE (
+        temp_table_in text,
+        files_in_default boolean,
```

Comment:
How about another beautification?
```diff
diff --git a/src/test/regress/input/temp_tablespaces.source b/src/test/regress/input/temp_tablespaces.source
index 25cab257231..5d50e7135f2 100644
--- a/src/test/regress/input/temp_tablespaces.source
+++ b/src/test/regress/input/temp_tablespaces.source
@@ -348,9 +348,7 @@ CREATE OR REPLACE FUNCTION gp_tablespace_file_report(
     n_temp_spill_files_tablespaces int
 ) RETURNS TABLE (
         temp_table_in text,
-        files_in_default boolean,
-        files_in_temp_tables boolean,
-        files_in_temp_files_n boolean
+        temp_files_in text
     ) AS $$
 DECLARE
     owo int;
@@ -409,9 +407,11 @@ BEGIN
     -- Check the existence of temporary hashjoin files.
     RETURN QUERY SELECT
         coalesce(temp_table_ts_name, 'default'),
-        gp_tablespace_watch_match(gp_execution_dbid(), 'default', '_HashJoin') > 0,
-        gp_tablespace_watch_match(gp_execution_dbid(), 'temp_tables', '_HashJoin') > 0,
-        gp_tablespace_watch_match(gp_execution_dbid(), session_file_ts_name, '_HashJoin') > 0;
+        case
+            when gp_tablespace_watch_match(gp_execution_dbid(), 'default', '_HashJoin') > 0 then 'default'
+            when gp_tablespace_watch_match(gp_execution_dbid(), 'temp_tables', '_HashJoin') > 0 then 'temp_tables'
+            when gp_tablespace_watch_match(gp_execution_dbid(), session_file_ts_name, '_HashJoin') > 0 then session_file_ts_name
+        end;
 END;
 $$ LANGUAGE plpgsql;
```

Reply:
How would we deal with varying tablespace number? Each session ID is unique, which will cause diff here.
https://github.com/arenadata/gpdb/blob/00a512d3d39f01053b40918d3e69ad6ecbc0d04d/src/test/regress/output/temp_tablespaces.source#L653-L666

Reply:
> How would we deal with varying tablespace number?

for example
```sql
when gp_tablespace_watch_match(gp_execution_dbid(), session_file_ts_name, '_HashJoin') > 0 then 'temp_files_n'
```